### ISSUE TITLE
curl_getinfo

### ISSUE THREAD

**Issue by tobciu** [linslin/Yii2-Curl#25]
please provide the informations as an associative array from curl_getinfo 

see: http://php.net/manual/de/function.curl-getinfo.php
-> 'If opt is given, returns its value. Otherwise, returns an associative array with the following elements (which correspond to opt)'

i need 'CURLINFO_REDIRECT_COUNT' for better parsing the response ;-)

**Comment by linslin** [comment]
Hey @tobciu , good one. This will be added!

**Comment by linslin** [comment]
Well, here you are. I released Version 1.0.6. 


```
        //init curl
        $curl = new curl\Curl();

        //get http://example.com/
        $response = $curl
            ->setOption(CURLOPT_CONNECTTIMEOUT, 5)
            ->setOption(CURLOPT_RETURNTRANSFER, true)
            ->get('http://www.linslin.org/');


        var_dump($curl->getInfo()); //return all infos as array
        var_dump($curl->getInfo(CURLINFO_REDIRECT_COUNT)); // return info by OPT
```

**Comment by tobciu** [comment]
in Line https://github.com/linslin/Yii2-Curl/blob/39218b57654d98f395acf050502a2becfaf98c97/Curl.php#L260 you can get a null-pointer exception if you call the 'reset' method before you call the 'getInfo' method

**Comment by linslin** [comment]
Thanks, is fixed in 1.0.7. Cheers.

**Comment by tobciu** [comment]
thank you for your fast response :+1: :-)
